controllers/finance_controller.py

```diff
@@ -1,26 +1,103 @@
-from tracker.finance_tracker import FinanceTracker
 from datetime import datetime
 
+from sqlalchemy import func, extract, and_
+from sqlalchemy.exc import IntegrityError
 
-class FinanceController:
+from database import SessionLocal
+from models.expense import Expense
+from models.investment import Investment
+from views.utilities import validate_date, validate_and_convert_date
+
+
+class FinanceTracker:
     def __init__(self):
-        self.finance_tracker = FinanceTracker()
+        # Initialize a new SQLAlchemy session
+        self.db_session = SessionLocal()
 
-    def add_expense(self, user_id, category, amount, date_str):
-        success, message = self.finance_tracker.add_expense(user_id, category, amount, date_str)
-        return success, message
+    def add_expense(self, user_id, category, amount, date):
```

Comment:
let's remove user_id, category, amount, date from the args
and make the Expense(data**) outside of this function, and pass it on as 

`    def add_expense(self, expense: Expense):`

Reply:
Yea, can we postpone this for now until we are certain about all the fields that the model(s) will carry. I envision adding some more fields or even changing totally the way the fields are handled based on future functionality. Advice